lib/formtastic/helpers/inputs_helper.rb

```diff
@@ -350,12 +350,21 @@ def association_columns(*by_associations) # @private
         end
       end
 
+      # Collects non-standard foreign key columns
+      def foreign_key_columns
+        if @object.present? && @object.class.respond_to?(:reflections)
+          @object.class.reflections.map {|_,v| v.options[:foreign_key] }.compact
```

Comment:
Shouldn't this use just `#foreign_key` method? Because I guess we don't want to show fields for associations with no options.

Also imo it should check just `ActiveRecord::Reflection::BelongsToReflection` as the other foreign keys will not exist on this table. That should be easily checked with `Reflection#belongs_to?`